ledger/store/src/helpers/rocksdb/internal/map.rs

```diff
@@ -277,17 +277,21 @@ impl<
 
         // Count the number of keys belonging to the map.
         let mut len = 0usize;
-        while let Some(key) = iter.key() {
-            // Only compare the map ID - the network ID is guaranteed to
-            // remain the same as long as there is more than a single map.
-            if key[2..][..2] != self.context[2..][..2] {
-                // If the map ID is different, it's the end of iteration.
+        while iter.valid() {
```

Comment:
What's the purpose of this change to the loop iteration?

As far as I can tell, the original loop: `while let Some(key) = iter.key()` should be equivalent but should avoid the redundant check? 

Is there additional validation somewhere that I'm missing?